File: Proxy Pattern/1. Minimal/Java/Main.java
public class Main {
    public static void main(String[] args) {
        System.out.println("Proxy Pattern in Java");
        System.out.println("---------------------");
        System.out.println();

        System.out.println("Create proxy instance.");
        AbstractSubject subject = new ProxySubject();

        System.out.println("Call proxy doSomething method.");
        subject.doSomething();
        System.out.println("Call proxy doSomething method again.");
        subject.doSomething();
    }
}

File: Proxy Pattern/1. Minimal/Python/main.py
from proxy_subject import ProxySubject

def main():
    print("Proxy Pattern in Python")
    print("-----------------------")
    print()

    print("Create proxy instance.")
    subject = ProxySubject()

    print("Call proxy doSomething method.")
    subject.doSomething()
    print("Call proxy doSomething method again.")
    subject.doSomething()

if __name__ == "__main__":
    main()
